sgkit/tests/test_popgen.py

```diff
@@ -68,18 +69,51 @@ def test_divergence(size, n_cohorts):
     ds = divergence(ds)
     div = ds.stat_divergence.values
 
+    # entries on the diagonal are diversity values
+    for i in range(n_cohorts):
+        ts_div = ts.diversity([subsets[i]], span_normalise=False)
+        np.testing.assert_allclose(div[i, i], ts_div)
+
+    # test off-diagonal entries, by replacing diagonal with NaNs
+    np.fill_diagonal(div, np.nan)
     ts_div = np.full([n_cohorts, n_cohorts], np.nan)
     for i, j in itertools.combinations(range(n_cohorts), 2):
         ts_div[i, j] = ts.divergence([subsets[i], subsets[j]], span_normalise=False)
-        ts_div[j, i] = ts_div[i, j]
+        ts_div[j, i] = ts.divergence([subsets[j], subsets[i]], span_normalise=False)
     np.testing.assert_allclose(div, ts_div)
 
 
+@pytest.mark.parametrize(
+    "size, n_cohorts",
+    [(2, 2), (3, 2), (10, 2), (100, 2)],
+)
+def test_Fst__Hudson(size, n_cohorts):
```

Comment:
Worth making chunking a part of the tests?

Reply:
Yes, definitely. Added a chunks parameter to all popgen tests. This exposed an issue in the divergence code, which I've now fixed.

https://github.com/pystatgen/sgkit/pull/302/commits/c9338ff8caa8deb6648aec957e7a277dfe7d293e